src/compiler/checker.ts

```diff
@@ -2067,37 +2067,38 @@ namespace ts {
                 // and we could then access that data during declaration emit.
                 writer.trackSymbol(symbol, enclosingDeclaration, meaning);
                 function walkSymbol(symbol: Symbol, meaning: SymbolFlags): void {
-                    if (symbol) {
-                        const accessibleSymbolChain = getAccessibleSymbolChain(symbol, enclosingDeclaration, meaning, !!(flags & SymbolFormatFlags.UseOnlyExternalAliasing));
-
-                        if (!accessibleSymbolChain ||
-                            needsQualification(accessibleSymbolChain[0], enclosingDeclaration, accessibleSymbolChain.length === 1 ? meaning : getQualifiedLeftMeaning(meaning))) {
+                    function climbSymbol(symbol: Symbol, meaning: SymbolFlags, endOfChain?: boolean): void {

```

Comment:
Can this be safely moved outside of the containing function? Normally we try to avoid nested functions because more allocations than needed